### PR TITLE
Bezier Line Data Provider

### PR DESCRIPTION
**Pull request by Railboy**
Overview
---
We've often found ourselves in need of a simpler / more lightweight alternative to the Spline data provider. This PR adds a Bezier line data provider and inspector.

bezier

### CODE REVIEW

File: Assets/MixedRealityToolkit/Utilities/Lines/DataProviders/BezierDataProvider.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Utilities.Lines.DataProviders
+{
+    public class BezierDataProvider : BaseMixedRealityLineDataProvider
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/Utilities/Lines/DataProviders/BezierDataProvider.cs]: Please decorate the class with a DataProvider attribute. Let me know if you need help deciding what values to use.  Thanks!

Comment (Railboy) [reply Assets/MixedRealityToolkit/Utilities/Lines/DataProviders/BezierDataProvider.cs]: The lines aren't associated with a service so I'm not sure what to use for serviceInterfaceType.
(Also would all the other line data providers need this attribute as well? Currently none of them do.)

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Utilities/Lines/DataProviders/BezierDataProvider.cs]: Good point. Let's chat offline and figure out a good solution.

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Utilities/Lines/DataProviders/BezierDataProvider.cs]: per our conversation, you shouldn't need the attribute if the class does not implement IMixedRealityDataProvider.

**Pull request merged**